drivers/wifi/winc1500/wifi_winc1500.c

```diff
@@ -1069,25 +1070,35 @@ static int winc1500_init(struct device *dev)
 	w1500_data.connected = false;
 
 	ret = m2m_wifi_init(&param);
-	if (ret) {
+	if (ret != M2M_SUCCESS) {
 		LOG_ERR("m2m_wifi_init return error!(%d)", ret);
 		return -EIO;
 	}
 
-	m2m_wifi_set_scan_region(WINC1500_REGION);
-
 	socketInit();
 	registerSocketCallback(winc1500_socket_cb, NULL);
 
-	m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid);
+	if (m2m_wifi_get_otp_mac_address(w1500_data.mac, &is_valid)) {
+		LOG_ERR("Failed set power profile");
```

Comment:
Log message is not correct, my fault, sorry.

Reply:
What do you want it to be?  "Failed to set MAC address"?

Reply:
I'll update it next push.

Reply:
@pabigot Please change to "Failed to get MAC address", thanks

Reply:
done